src/views/DatasetManager/components/DeleteDatasetButton/DeleteDatasetButton.js

```diff
@@ -0,0 +1,75 @@
+import { React, useCallback } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import DeleteForeverIcon from '@mui/icons-material/DeleteForever';
+import { IconButton } from '@mui/material';
+import { INGESTION_STATUS } from '../../../../services/config/ApiConstants';
+import { TwoActionsDialogService } from '../../../../services/twoActionsDialog/twoActionsDialogService';
+import { useDeleteDatasetButton } from './DeleteDatasetButtonHooks';
+
+export const DeleteDatasetButton = ({ dataset, location }) => {
+  const isDisabled = () => {
+    if (dataset.ingestionStatus === INGESTION_STATUS.PENDING) {
+      return true;
+    } else {
+      return false;
+    }
+  };
```

Comment:
This function can be replaced by a simple bool value, that will then be used in the component render.
This would be simply declared with: `const isDisabled = dataset.ingestionStatus === INGESTION_STATUS.PENDING;`
And in the render:
  `disabled={isDisabled}`
and
  `<DeleteForeverIcon color={isDisabled ? 'disabled' : 'primary'} />`